api/src/main/scala/hmda/api/protocol/processing/MsaProtocol.scala

```diff
@@ -1,11 +1,36 @@
 package hmda.api.protocol.processing
 
-import hmda.query.model.filing.{ Irs, Msa, MsaSummary, MsaWithName }
-import spray.json.DefaultJsonProtocol
+import hmda.api.model.{ IrsResponse, PaginatedResponse }
+import hmda.query.model.filing.{ MsaSummary, MsaWithName }
+import spray.json._
 
-trait MsaProtocol extends DefaultJsonProtocol {
-  implicit val msaProtocol = jsonFormat13(Msa.apply)
+trait MsaProtocol extends DefaultJsonProtocol with ParserResultsProtocol {
   implicit val msaWithNameProtocol = jsonFormat14(MsaWithName.apply)
   implicit val msaSummaryProtocol = jsonFormat12(MsaSummary.apply)
-  implicit val irsProtocol = jsonFormat2(Irs.apply)
+
+  implicit object IrsResponseJsonFormat extends RootJsonFormat[IrsResponse] {
+    override def write(irs: IrsResponse): JsValue = {
+      JsObject(
+        "msas" -> irs.msas.toJson,
+        "summary" -> irs.summary.toJson,
+        "count" -> JsNumber(irs.count),
+        "total" -> JsNumber(irs.total),
+        "_links" -> irs.links.toJson
+      )
+    }
+
+    override def read(json: JsValue): IrsResponse = {
+      json.asJsObject.getFields("msas", "summary", "total", "_links") match {
+        case Seq(JsArray(msas), JsObject(summ), JsNumber(tot), JsObject(links)) =>
+          val msaCollection: Seq[MsaWithName] = msas.map(_.convertTo[MsaWithName])
+          val path: String = PaginatedResponse.staticPath(links("href").convertTo[String])
+          val currentPage: Int = PaginatedResponse.currentPage(links("self").convertTo[String])
+          val summary: MsaSummary = JsObject(summ).convertTo[MsaSummary]
+          val total: Int = tot.intValue
+          IrsResponse(msaCollection.toList, summary, path, currentPage, total)
```

Comment:
I would remove the types for these `val` declarations if the compiler is happy with it. In general, `val` or `var` declarations that can use type inference should do so.